test/test_rake_task.rb

```diff
@@ -296,24 +296,26 @@ def test_all_prerequisite_tasks_includes_self_on_cyclic_dependencies
     assert_equal [a, b], a.all_prerequisite_tasks.sort_by { |t| t.name }
   end
 
-  def test_timestamp_returns_now_if_all_prereqs_have_no_times
+  def test_timestamp_returns_now_regardless_prereq_times
     a = task a: ["b", "c"]
-    task :b
-    task :c
+    b = task :b
+    c = task :c
+
+    def b.timestamp() Time.now + 10 end
```

Comment:
I change the dependencies timestamp here to ensure they do not affect the dependent task timestamp.